routes/guests.rb

```diff
@@ -0,0 +1,69 @@
+class Guests < Cuba
+  define do
+    # Homepage for campaigners (when going to http://192.168.59.103:5000/actionite)
+    on "actionite" do
+      render("actionite", title: "Actionite | Login")
+    end
+
+    # Here is where the Google API first response arrives and we send it
+    # to "login/:access_token"
+    on "google_oauth" do
+      on param("access_token") do |access_token|
+        res.redirect "/login/#{ access_token }"
+      end
+    end
+
+    # Once a campaigner clicks on SignIn google button, we send the request to
+    # Google and receive several values from the Google API. Then we use the
+    # 'token_access' value to send a GET request to Google again to get some
+    # user's details, google id and email among others. We use those values to
+    # create a new campaigner in the db (without any password), then we
+    # authenticate that campaigner and then it redirects to '/dashboard'
+    # on Campaigners route which will be the homepage for the campaigns.
+    on "login/:access_token" do |access_token|
+      google_user = Google.fetch_user(access_token)
+
+      # Example: {"kind"=>"plus#person", "etag"=>"\"RqKWnRU4WW46-6W3rWhLR9iFZQM/kcT6GjinXWesHqvTtumowMTYI8k\"", "gender"=>"female", "emails"=>[{"value"=>"cecilia@sumofus.org", "type"=>"account"}], "objectType"=>"person", "id"=>"100501244795726503142", "displayName"=>"Cecilia Rivero", "name"=>{"familyName"=>"Rivero", "givenName"=>"Cecilia"}, "url"=>"https://plus.google.com/100501244795726503142", "image"=>{"url"=>"https://lh3.googleusercontent.com/-XdUIqdMkCWA/AAAAAAAAAAI/AAAAAAAAAAA/4252rscbv5M/photo.jpg?sz=50", "isDefault"=>true}, "isPlusUser"=>true, "circledByCount"=>0, "verified"=>false, "domain"=>"sumofus.org"}
```

Comment:
A nice to have for this would be saving the user's image URL as well and loading that when they're logged in to say "You're logged in as [image] $NAME" on a top bar or something.